### PR TITLE
neocities: remove package duplication

### LINKED ISSUE #330321: Duplicate Package: neocities & neocities-cli
I'm not really sure how it happened, however we have 2 copies of the Neocities CLI: One from #250985, and another from #290514. After formatting the nix files in each, the only differences are:
- Description
- Maintainers
- Platforms (`neocities-cli` lists `platforms.unix`)
- Gemfile lock platform/bundled-with version (`neocities` is bundled-with `2.1.4`, `neocities-cli` has `x86_64-linux` added as a platform and is bundled-with  `2.5.5`)

`neocities` is older, while `neocities-cli` is newer and slightly more complete, as it seems.

One should be removed, with the other aliased to it.

### Notify maintainers

CC @dawoox as the maintainer of the other Neocities package.

### PR DESCRIPTION
**Pull request by stupidcomputer**
## Description of changes

Pursuant to #330321, `neocities` and `neocities-cli` are duplicates of each other. This MR de-duplicates them. ~~I've chosen to stick with the `neocities-cli` package because it's newer.~~
*See discussion below -- the `neocities-cli` package is moved to `neocities`.*

Closes #330321.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Infinidoge** [comment]
I don't use Neocities anymore, so go ahead and remove the commit adding me as a maintainer

**Review by Infinidoge — APPROVED** [review]
Outside of removing me as a maintainer, looks good to me!

**Comment by stupidcomputer** [comment]
@Infinidoge, you've been removed. Thanks!

**Comment by philiptaron** [comment]
Super small nit.

https://github.com/neocities/neocities-ruby is both a cli and a library. It's currently not depended-on by anything in nixpkgs, but if it were, I'd prefer the `neocities` name. By virtue of adding `neocities` to `aliases.nix`, rather than `neocities-cli`, if another package in nixpkgs wants to depend on it, they have to use the `neocities-cli` name.

So I'd be in favor of moving the `neocities-cli` derivation over to `pkgs/by-name/ne/neocities` and making the `-cli` name the alias.

**Comment by stupidcomputer** [comment]
@philiptaron, you're correct. Your revisions have been incorporated into the latest push. Thanks!

**Review by philiptaron — APPROVED** [review]
Thanks! ❤️

File: pkgs/by-name/ne/neocities/package.nix [hunk 1]
```diff
@@ -10,10 +11,12 @@ bundlerApp {
   passthru.updateScript = bundlerUpdateScript "neocities";
 
   meta = with lib; {
-    description = "The Neocities CLI, written in Ruby";
+    description = "Neocities Gem - A CLI and library for using the Neocities web site API";
```
Comment (Sigmanificient) [inline pkgs/by-name/ne/neocities/package.nix:14]: Small nit:

`meta.description` must: "Not start with the package name" 
> https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ne/neocities/package.nix (hunk 1 above)
Comment (stupidcomputer) [reply pkgs/by-name/ne/neocities/package.nix]: I've revised it to read "cli and library for interacting with the Neocities API" -- is this okay?

Comment (Sigmanificient) [reply pkgs/by-name/ne/neocities/package.nix]: I believe it is

File: pkgs/by-name/ne/neocities/package.nix [hunk 2]
```diff
@@ -10,10 +11,12 @@ bundlerApp {
   passthru.updateScript = bundlerUpdateScript "neocities";
 
   meta = with lib; {
-    description = "The Neocities CLI, written in Ruby";
+    description = "cli and library for interacting with the Neocities API";
```
Comment (philiptaron) [inline pkgs/by-name/ne/neocities/package.nix:14]:
```suggestion
    description = "CLI and library for interacting with the Neocities API";
```

**Review by philiptaron — APPROVED** [review]
# :shipit:

File: pkgs/by-name/ne/neocities/package.nix (hunk 2 above)
Comment (stupidcomputer) [reply pkgs/by-name/ne/neocities/package.nix]: I've incorporated your suggestions, thanks!

**Review by philiptaron — APPROVED** [review]

**Pull request merged**